corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -134,6 +135,17 @@ def update_sync_log_with_checks(sync_log, xform, cases, case_db,
                                         case_id_blacklist=case_id_blacklist)
 
 
+def prune_previous_log(sync_log):
```

Comment:
This adds three queries to couch for every form submission. Should we attempt to batch these and perform deletions asynchronously?